hack/setup-cluster.sh

```diff
@@ -543,7 +552,10 @@ load() {
   fi
 
   echo "${bright}Building operator from current worktree${reset}"
-
+  
+  if [[ "$OS" == "darwin" && "$ARCH" == "arm64" ]]; then
+    export GORELEASER_CURRENT_TAG=v0.0.0
+  fi
```

Comment:
I do not get the reason for this change.

Reply:
without it my build straight up fails :(
```
Schwerwiegend: Keine Namen gefunden, kann nichts beschreiben.
GOOS=linux GOARCH=arm64 GOPATH= DATE=2023-10-05 COMMIT=0958e371 VERSION= \
	  /Users/lkrauch/inovex/scrumlr/cloudnative-pg/bin/goreleaser build --skip-validate --clean --single-target
  • starting build...
  • loading config file                              file=.goreleaser.yml
  • building only for linux/arm64                    reason=single target is enabled
  • loading environment variables
  • getting and validating git state
  ⨯ build failed after 0s                    error=git doesn't contain any tags. Either add a tag or use --snapshot
make[1]: *** [docker-build] Error 1
make: *** [deploy-locally] Error 2
```

Reply:
That's because you cloned the repository without fetching the tags.

Reply:
Couldnt find anything related to this in the setup doc. Maybe worth adding?

Reply:
Feel free to propose a patch 😄

Reply:
https://github.com/cloudnative-pg/cloudnative-pg/pull/2247 may be helpful here too

Reply:
I merged #2247, which fixes this issue.